src/buttons/help.ts

```diff
@@ -0,0 +1,40 @@
+import { ButtonBuilder, ButtonStyle } from "discord.js";
+import type InteractionContext from "../models/InteractionContext";
+import NeedleButton from "../models/NeedleButton";
+import type NeedleBot from "../NeedleBot";
+import ObjectFactory from "../ObjectFactory";
+import type CommandExecutorService from "../services/CommandExecutorService";
+
+export default class HelpButton extends NeedleButton {
+	public readonly customId = "help";
+	private readonly commandExecutor: CommandExecutorService;
+
+	constructor(bot: NeedleBot) {
+		super(bot);
+		this.commandExecutor = ObjectFactory.createCommandExecutorService();
```

Comment:
I tried figuring out a solution for that, but I couldn't get it any closer to constructor DI than this. We need to use a service locator pattern *somewhere*, and I wasn't keen on having the command loader figure out all the required parameters for every command constructor, because I don't even think that's possible without decorators. And I'm not brave enough for decorators yet.